include/deal.II/numerics/vector_tools.h

```diff
@@ -969,13 +974,14 @@ namespace VectorTools
    * apply as for the previous function, in particular about the use of the
    * component mask and the requires size of the function object.
    */
-  template <typename DoFHandlerType, typename number>
+  template <int dim, int spacedim, template <int,int> class DoFHandlerType,
```

Comment:
I am not sure that I understand what you are asking. I believe the answer is 'no' since then `dim` would not show up anywhere in the function signature.